hexrd/ui/interactive_template.py

```diff
@@ -27,11 +23,48 @@ def __init__(self, parent=None):
         self.translation = [0, 0]
         self.complete = False
         self.event_key = None
-        self.parent.setFocusPolicy(Qt.ClickFocus)
+        self.detector = detector
+        self.instrument = instrument
+        self._static = True
+
+        for canvas in self.image_tab_widget.active_canvases:
+            canvas.setFocusPolicy(Qt.ClickFocus)
+
+    @property
+    def current_canvas(self):
+        idx = self.image_tab_widget.current_index
+        return self.image_tab_widget.active_canvases[idx]
+
+    @property
+    def ax(self):
+        idx = HexrdConfig().current_imageseries_idx
+        return self.current_canvas.axes_images[idx]
 
     @property
     def raw_axes(self):
-        return list(self.parent.raw_axes.values())[0]
+        if not self.current_canvas.raw_axes:
+            return self.current_canvas.axis
+
+        for axes in self.current_canvas.raw_axes.values():
+            if axes.get_title() == self.detector:
+                return axes
+
+        return list(self.current_canvas.raw_axes.values())[0]
+
+    @property
+    def static_mode(self):
+        return self._static
+
+    @static_mode.setter
+    def static_mode(self, mode):
+        if mode == self._static:
+            return
+
+        if mode:
+            self.disconnect()
+        else:
+            self.connect_translate_rotate()
```

Comment:
Just to be overly code-defensive (and prevent potential bugs), can we call `disconnect()` at the beginning of `self.connect_translate_rotate()` to ensure any previous connections are disconnected?

Additionally, we can make sure `disconnect()` can be safely called if 1. no connections were made or 2. `disconnect()` has already been called once before.

Reply:
Done